test/common/router/config_impl_test.cc

```diff
@@ -279,6 +284,23 @@ TEST(RouteMatcherTest, TestRoutes) {
                                                 Http::LowerCaseString("x-envoy-virtual-cluster")}),
               ContainerEq(config.responseHeadersToRemove()));
 
+  // Request header manipulation testing
+  {
+    // At route_config level
+    // Check if user-specified request headers are added to config before scrubbing
+    EXPECT_THAT((std::list<std::pair<Http::LowerCaseString, std::string>>(
+                    {{Http::LowerCaseString("x-backend-mystery-header"), "true"},
+                     {Http::LowerCaseString("Connection"), "Keep-Alive"}})),
+                ContainerEq(config.requestHeadersToAdd()));
+
+    Http::TestHeaderMapImpl headers = genHeaders("api.lyft.com", "/host/rewrite/me", "GET");
```

Comment:
finalizeRequestHeaders doesn't invoke the added code for `request_headers_to_add`. Please add a test to https://github.com/lyft/envoy/blob/master/test/common/http/conn_manager_utility_test.cc